### ISSUE TITLE
FacebookSDK is outdated

### ISSUE THREAD

**Issue by tillsanders** [NativeScript/nativescript-facebook#233]
If I'm not mistaken, the Facebook SDK is set to v5.0, while the [current version is 8.2](https://github.com/facebook/facebook-ios-sdk/releases). So far, I'm not experiencing troubles, but this is probably only a matter of time.

Is this plugin still maintained, or is there a reason, why it has not upgraded beyond v5.0?

**Comment by StefanNedelchev** [comment]
@tillsanders  Indeed, the current SDK is not only outdated but it's also officially deprecated and they will stop accepting API calls until 2023. I'm also surprised that this plugin hasn't been updated for so long. Sure, it's working just fine and that's probably a reason. Maybe they continued the support in pro plugins, who knows?

**Comment by tillsanders** [comment]
There is no ProPlugin, but @bradmartin said on Slack he would take a look when he has some time. But of course, we all know how difficult it can be to make some room for stuff that's not the top priority!

**Comment by NathanaelA** [comment]
1. It is **all** open source software, nobody on the team is paid to fix the NativeScript plugins or work on NativeScript directly UNLESS it is a support contract request of a bug report against it.  The majority of the work down on plugins and NativeScript is done for _free_ by the core team because we love the framework.   So we have to prioritize what gets done frequently, as there is always more work than workers and issues that effect the majority of the community has a higher priority over a plugin that still works but should be upgraded at some point.  

2. ProPlugins has not ever used any plugins from the NativeScript group, in fact the opposite is true; NativeScript has had a couple plugins merged into the main Core from ProPlugins, and there is a RFC's for another two I would like to move into Core as I feel they belong in core more than as a separate plugin.

3. In the grand scheme of things the Facebook plugin is much lower on the priority list, as none of our current projects need it to my knowledge.   This can also play a role in when plugins get upgraded.  With that said:
- We love support contracts (which indirectly fund things like this, as support work is paid work on the framework)
- Paid contracts/jobs (we can be hired to work specifically on any plugin for you and your company)
- Community PR's to upgrade it or even PR's to any of the other repos including DOC's to fix ANY issues.
- Answering issues like this with answers, so that the core team doesn't have to.

Anything that can either help fund it, or can reduce the load for those who are volunteering their time to keep the framework going forward has an impact on the entire project, including all the plugins.   We all play a part in moving the NativeScript framework forward, the more people who choose to get involved and move it forwards makes the burden on everyone else already contributing lighter.   Which then makes it easier to find the time to work on pieces that don't see as much love...

**Comment by tillsanders** [comment]
@NathanaelA Sorry if this came across wrong! I didn't mean to imply any obligation for anyone to put his/her time in this, because – as you said – it's open source and everyone has their own priorities.

**Comment by tillsanders** [comment]
I just learned that you have to upgrade the Facebook SDK to use tracking on iOS 14. I think this impacts the priority of this issue, so I just wanted to leave this information here, should anyone be willing to work on this.

![image006](https://user-images.githubusercontent.com/1533846/108827479-f5561a80-75c5-11eb-96cf-4c3c6f0ab9fa.jpg)

**Comment by StefanNedelchev** [comment]
> I just learned that you have to upgrade the Facebook SDK to use tracking on iOS 14. I think this impacts the priority of this issue, so I just wanted to leave this information here, should anyone be willing to work on this.
> 
> ![image006](https://user-images.githubusercontent.com/1533846/108827479-f5561a80-75c5-11eb-96cf-4c3c6f0ab9fa.jpg)

Facebook are ending their tracking service in June so if that's your only concern - you won't have to worry about it very soon :laughing:

**Comment by tillsanders** [comment]
> Facebook are ending their tracking service in June so if that's your only concern - you won't have to worry about it very soon 😆

I think there might be a misconception – they are ending Facebook Analytics, but not the tracking itself. Analytics was just a portal to analyse the collected data. But yeah, nonetheless, tracking is about to take a big hit with iOS 14.5 – thankfully!

**Comment by StefanNedelchev** [comment]
I submitted a PR which updates the SDK for iOS since it's critical right now - iOS 14+ requires Facebook SDK 8 or later for advertising related stuff to work https://github.com/NativeScript/nativescript-facebook/pull/235

**Comment by update-switzerland** [comment]
For those struggling with this: Just update the plugin's podfile to use SDK v. 11.0.0

```
pod 'FBSDKCoreKit', '~> 11.0.0'
pod 'FBSDKLoginKit', '~> 11.0.0'
pod 'FBSDKShareKit', '~> 11.0.0'

```
